src/components/HdToggle.vue

```diff
@@ -0,0 +1,251 @@
+    },
+    canBeToggled: {
+      type: Boolean,
+      default: true,
+    },
+    open: {
+      type: Boolean,
+      default: true,
+    },
+    transitionDuration: {
+      type: Number,
+      default: 300,
+    },
+  },
+  data() {
+    return {
+      // bodyHeight is a string only initially, will be set to a Number
+      // as soon as it is mounted
+      bodyHeight: '100%',
+      isUsingMouse: false,
+      internalFocusRemoved: false,
+      chevronIcon,
+    };
+  },
+  computed: {
+    maxHeight() {
+      if (this.open) {
+        return this.bodyHeight;
+      }
+
+      return 0;
+    },
+  },
+  watch: {
+    open: {
+      handler(open) {
+        this.ensureBodyScrolledToTop();
+
+        this.$nextTick(() => {
+          if (open) {
+            this.enableInternalFocus();
+          } else {
+            this.disableInternalFocus();
+          }
+        });
+      },
+      immediate: true,
+    },
+  },
+  mounted() {
+    this.$nextTick(this.calculateBodyHeight);
+    this.addResizeEvents();
+  },
+  beforeDestroy() {
+    this.removeResizeEvents();
+  },
+  methods: {
+    // Encapsulated in a method to be able to mock it in the tests
+    getScrollHeight(el) {
+      return el.scrollHeight;
+    },
+    toggleOpen() {
+      if (this.canBeToggled === false) {
+        return;
+      }
+
+      this.$emit('toggle', !this.open);
+    },
+    calculateBodyHeight() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      this.bodyHeight = this.getScrollHeight(this.$refs.body);
+    },
+    ensureBodyScrolledToTop() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      this.$refs.body.scrollTop = 0;
+    },
+    disableInternalFocus() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      const $innerFocusableElements = this.$refs.body.querySelectorAll(
+        'a, button, [tabindex]',
+      );
+
+      if (!$innerFocusableElements.length) {
+        return;
+      }
+
+      Array.from($innerFocusableElements).forEach((el) => {
+        const currentTabIndex = el.getAttribute('tabindex');
+
+        if (currentTabIndex !== null) {
+          el.setAttribute('data-hd-tabindex-backup', currentTabIndex);
+        }
+
+        el.setAttribute('tabindex', '-1');
+      });
+
+      this.internalFocusRemoved = true;
+    },
+    enableInternalFocus() {
+      if (!this.$refs.body || this.internalFocusRemoved === false) {
+        return;
+      }
+
+      const $innerFocusableElements = this.$refs.body.querySelectorAll(
+        'a, button, [tabindex]',
+      );
+
+      if (!$innerFocusableElements.length) {
+        return;
+      }
+
+      Array.from($innerFocusableElements).forEach((el) => {
+        const oldTabIndex = el.getAttribute('data-hd-tabindex-backup');
+
+        if (oldTabIndex !== null) {
+          el.setAttribute('tabindex', oldTabIndex);
+          el.removeAttribute('data-hd-tabindex-backup');
+        } else {
+          el.removeAttribute('tabindex');
```

Comment:
Here you might end up removing the a `tabidex` that wasn't set by this component.
I'd suggest that in this method instead of querying this `'a, button, [tabindex]'` query `'[data-hd-tabindex-backup]'`, so this way we get just the elements we changed, and the logic will be also simplified as you know that all the elements you get have this attribute (so you just set the old and remove the backup).